lib/Dialect/SV/Transforms/SVExtractTestCode.cpp

```diff
@@ -331,14 +333,43 @@ static void addExistingBinds(Block *topLevelModule, BindTable &bindTable) {
 }
 
 // Inline any modules that only have inputs for test code.
-static void inlineInputOnly(hw::HWModuleOp oldMod,
-                            hw::InstanceGraph &instanceGraph,
-                            BindTable &bindTable,
-                            SmallPtrSetImpl<Operation *> &opsToErase) {
+static void
+inlineInputOnly(hw::HWModuleOp oldMod, hw::InstanceGraph &instanceGraph,
+                BindTable &bindTable, SmallPtrSetImpl<Operation *> &opsToErase,
+                DenseSet<hw::InnerRefAttr> &innerRefUsedByNonBindOp) {
+
   // Check if the module only has inputs.
   if (oldMod.getNumOutputs() != 0)
     return;
 
+  // Check if it's ok to inline. We cannot inline the module if there exists a
+  // declaration with an inner symbol referred by non-bind ops (e.g. hierpath).
+  for (auto port : oldMod.getAllPorts()) {
+    if (port.sym) {
+      if (!port.sym.getSymName())
+        return;
+      if (innerRefUsedByNonBindOp.count(hw::InnerRefAttr::get(
+              oldMod.getModuleNameAttr(), port.sym.getSymName()))) {
+        emitWarning(oldMod.getLoc()) << "module " << oldMod.getModuleName()
+                                     << " is an input only module but cannot "
+                                        "be inlined because a signal "
+                                     << port.name << " is referred by names";
```

Comment:
This error message is great, but I got confused at the very end with `names` (plural). I think it makes sense to just say `referred by name`, was that the intent?